src/logging.cpp

```diff
@@ -282,9 +286,6 @@ void BCLog::Logger::LogPrintStr(const std::string& str)
         fwrite(str_prefixed.data(), 1, str_prefixed.size(), stdout);
         fflush(stdout);
     }
-    for (const auto& cb : m_print_callbacks) {
-        cb(str_prefixed);
-    }
     if (m_print_to_file) {
```

Comment:
It may be better to have entries for `m_print_to_file` (and also `m_print_to_console` above) in the `m_print_callbacks` list, rather than having them here as special cases.  I may investigate that approach, and if it seems reasonable, make a separate PR (to merge ahead of the current PR).